### DISCUSSION TITLE
[Formily ArrayTable] 表单场景如何自定义button，和获取某一行数据

### DISCUSSION THREAD

**Discussion by jxp1147** [alibaba/formily#2573] [Q&A 2.x] [answered]
[Formily ArrayTable] 表单场景如何自定义button，和获取某一行数据

**Comment by janryWang** [comment, marked as answer]
```
import {ArrayTable} from '@formily/antd'

createSchemaField({
   components:{
      MyButton:()=>{
        const record = ArrayTable.useRecord()
        return <Button>xxx</Button>
      }
   }
})

```

**Comment by jxp1147** [comment]
谢谢


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "alibaba/formily"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月6日(星期一) 中午11:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [alibaba/formily] [Formily ArrayTable] 表单场景如何自定义button，和获取某一行数据 (Discussion #2573)





 import {ArrayTable} from ***@***.***/antd' createSchemaField({    components:{       MyButton:()=&gt;{         const record = ArrayTable.useRecord()         return <Button&gt;xxx</Button&gt;       }    } })  
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jxp1147** [comment]
你好，ArrayTable.useRecord这个方法在开发环境中是没问题的，但是项目打包之后，输出ArrayTable.useRecord为undefined


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "alibaba/formily"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月6日(星期一) 中午11:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [alibaba/formily] [Formily ArrayTable] 表单场景如何自定义button，和获取某一行数据 (Discussion #2573)





 import {ArrayTable} from ***@***.***/antd' createSchemaField({    components:{       MyButton:()=&gt;{         const record = ArrayTable.useRecord()         return <Button&gt;xxx</Button&gt;       }    } })  
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by relines** [comment]
解决了么？
